### PR TITLE
docs: drink

### PR DESCRIPTION
**Pull request by Daanvdplas**
I would suggest to read the docs by getting it locally and see how it reads and if everything is clear. I changed the naming back to `ApiError` as I think it was the correct name. It refers to the error type in the api. My apologies for that...

I removed many of the readme docs as many was already covered somewhere else. Not repeating ourselves minimises the amount of code to maintain and chances for small mistakes.

Please read the code examples provided carefully as I found mistakes in your PR. Imagine you looking into a new library and you find out there are mistakes in the example docs? To me this wouldn't be a good sign. Therefore another reason to keep things simple and short (often also a sign the code is well understood :) ).

Please make final changes in a single commit (so I can check) and merge into your PR. Then, `feat-runtime_error` can be merged too. Your readme PR won't be necessary if you agree with these changes.

### CODE REVIEW

File: crates/pop-drink/src/lib.rs [hunk 1]
```diff
@@ -6,91 +8,64 @@ use scale::Decode;
...
 
+/// Error type and utilities for testing contracts using the Pop API.
 pub mod error;
 #[cfg(test)]
 mod mock;
 
+/// Types and utilities for testing smart contracts interacting with Pop Network Devnet via the pop api.
 pub mod devnet {
-	pub use pop_runtime_devnet::Runtime;
+	pub use pop_runtime_devnet::{RuntimeError::*, Runtime};
```
Comment (chungquantin) [inline crates/pop-drink/src/lib.rs:18]: Conflict exports. `Session` is exported here, while the `Session` type for testing is exported in the next line.

File: crates/pop-drink/src/error.rs [hunk 2]
```diff
@@ -1,63 +1,181 @@
...
+/// - `result`: The result which contains the custom error type.
+/// - `error`: The expected error.
+#[macro_export]
+macro_rules! assert_err {
+    ($result:expr, $error:expr $(,)?) => {
+        $crate::error::assert_err_inner::<_, _, _>($result, $error);
+    };
+}
+
 #[track_caller]
```
Comment (chungquantin) [inline crates/pop-drink/src/error.rs:177]:
```suggestion
#[track_caller]
#[allow(unused)]
```

File: crates/pop-drink/src/error.rs [hunk 3]
```diff
@@ -1,63 +1,181 @@
...
+/// 			...,
+/// 			CustomError::StatusCode(status_code) => status_code,
+/// 		}
+/// 	}
+/// }
+///
+/// - Use `assert_err` in a test.
+///
+/// #[drink::test(sandbox = Pop)]
+/// fn test_custom_error(mut session: Session) {
```
Comment (chungquantin) [inline crates/pop-drink/src/error.rs:150]: Thanks for this example code, looks better.

File: crates/pop-drink/src/error.rs [hunk 4]
```diff
@@ -1,63 +1,181 @@
...
+/// 	...
+///
+/// 	// Call a contract method that returns a `Result<(), CustomError>`.
+///		let result = call::<Pop, (), CustomError>(session, "hello_world", vec![], None);
```
Comment (chungquantin) [inline crates/pop-drink/src/error.rs:154]: Need to reformat. Wrong indentation.

File: crates/pop-drink/src/error.rs [hunk 5]
```diff
@@ -1,63 +1,181 @@
...
-/// - `Error` - `Error` to assert against the custom error type.
+/// - Create a custom error type that holds the status code.
+///
+/// ```rs
+/// use pop_api::StatusCode;
 ///
-/// # Parameters
+/// /// Custom error in contract.
+/// pub enum CustomError {
+///		...,
```
Comment (chungquantin) [inline crates/pop-drink/src/error.rs:120]: Wrong indentation.

**Review by chungquantin — CHANGES_REQUESTED** [review]
There are a few nitpicks that need to tackle. Aside from that, I like your doc. It is clean to me. Thanks for making these changes .

**Comment by Daanvdplas** [comment]
Thank you, resolved all your comments!

**Pull request merged**